dynamic-check/src/objects/frame.rs

```diff
@@ -0,0 +1,1198 @@
+        // Parse all statements.
+        for statement in statements.iter() {
+            self.parse_statement(statement)?;
+        }
+
+        // Pop out of scope.
+        let _scope = self.pop_scope();
+
+        Ok(())
+    }
+
+    ///
+    /// Collects `TypeAssertion` predicates from a conditional statement.
+    ///
+    /// Creates a new scope for each code block in the conditional.
+    ///
+    fn parse_statement_conditional(
+        &mut self,
+        conditional: &ConditionalStatement,
+        span: &Span,
+    ) -> Result<(), FrameError> {
+        // Parse the condition expression.
+        let condition = self.parse_expression(&conditional.condition)?;
+
+        // Assert that the condition is a boolean type.
+        let boolean_type = Type::Boolean;
+        self.assert_equal(boolean_type, condition, span);
+
+        // Parse conditional statements.
+        self.parse_block(&conditional.statements, span)?;
+
+        // Parse conditional or end.
+        match &conditional.next {
+            Some(cond_or_end) => self.parse_conditional_nested_or_end(cond_or_end, span)?,
+            None => {}
+        }
+
+        Ok(())
+    }
+
+    ///
+    /// Collects `TypeAssertion` predicates from a conditional statement.
+    ///
+    fn parse_conditional_nested_or_end(
+        &mut self,
+        cond_or_end: &ConditionalNestedOrEndStatement,
+        span: &Span,
+    ) -> Result<(), FrameError> {
+        match cond_or_end {
+            ConditionalNestedOrEndStatement::Nested(nested) => self.parse_statement_conditional(nested, span),
+            ConditionalNestedOrEndStatement::End(statements) => self.parse_block(statements, span),
+        }
+    }
+
+    ///
+    /// Collects `TypeAssertion` predicates from an iteration statement.
+    ///
+    fn parse_iteration(
+        &mut self,
+        identifier: &Identifier,
+        from_to: &(Expression, Expression),
+        statements: &Vec<Statement>,
+        span: &Span,
+    ) -> Result<(), FrameError> {
+        // Insert variable into symbol table with u32 type.
+        let u32_type = Type::IntegerType(IntegerType::U32);
+        let _expect_none = self.insert_variable(identifier.name.to_owned(), u32_type.clone(), span);
+
+        // Parse `from` and `to` expressions.
+        let from_type = self.parse_expression(&from_to.0)?;
+        let to_type = self.parse_expression(&from_to.1)?;
+
+        // Assert `from` and `to` types are a u32 or implicit.
+        self.assert_equal(u32_type.clone(), from_type, span);
+        self.assert_equal(u32_type, to_type, span);
+
+        // Parse block of statements.
+        self.parse_block(statements, span)
+    }
+
+    ///
+    /// Asserts that the statement `UnresolvedExpression` returns an empty tuple.
+    ///
+    fn parse_statement_expression(&mut self, expression: &Expression, span: &Span) -> Result<(), FrameError> {
+        // Create empty tuple type.
+        let expected_type = Type::Tuple(Vec::with_capacity(0));
```

Comment:
```suggestion
        let expected_type = Type::Tuple(Vec::new());
```
`Vec::new()` doesn't allocate